x-pack/plugins/security_solution/public/exceptions/hooks/use_endpoint_exceptions_capability/index.tsx

```diff
@@ -6,18 +6,24 @@
  */
 
 import { useMemo } from 'react';
-import { useListsConfig } from '../../../detections/containers/detection_engine/lists/use_lists_config';
+import { useListsPrivileges } from '../../../detections/containers/detection_engine/lists/use_lists_privileges';
 import { useHasSecurityCapability } from '../../../helper_hooks';
+import { useKibana } from '../../../common/lib/kibana';
 
 export const useEndpointExceptionsCapability = (
   capability: 'showEndpointExceptions' | 'crudEndpointExceptions'
-) => {
-  const { loading: listsConfigLoading, needsConfiguration: needsListsConfiguration } =
-    useListsConfig();
+): boolean => {
+  const { lists } = useKibana().services;
+  const { canManageIndex, loading: privilegesLoading } = useListsPrivileges();
+  const enabled = lists != null;
+  const needsIndexConfiguration = canManageIndex === false;
+  const needsConfiguration = !enabled || needsIndexConfiguration;
+  const hasAccessToLists = !(privilegesLoading || needsConfiguration);
```

Comment:
This looks like it was a partial copy/paste from `useListsConfig`, e.g. `needsConfiguration` had a more comprehensive definition originally, and as such isn't quite correct here. But that logic was also specific to creating the lists index, and I think you could eliminate it entirely here (it doesn't matter whether they _could_ create the index, we're not trying to do that here):

```suggestion
  const listsEnabled = lists != null;
  const hasListsAccess = canManageIndex;
  const canUseLists = !privilegesLoading && listsEnabled && hasListsAccess
```

However, I had a question about the definition of `hasListsAccess`, since the above shows more clearly that this is centered around the `canManageIndex` privilege: If this capability requires reading/writing lists data, then `canReadIndex` and `canWriteIndex` from `useListsPrivileges` are what you want, here: `manage` has no relation to that concept.

Reply:
Thanks for looking into this @rylnd. I do think using `canReadIndex` and `canWriteIndex` makes more sense than the manage privilege. I'll make that change.

Reply:
9a079cf3dd57b6d2a36070a40de7e2cce479d14e

Reply:
and 2c0203306b529a4235a44fd7a70f126f2cd44967

Reply:
@rylnd @kqualters-elastic @yctercero I'm now wondering if there's a way to verify index privileges on the server side for API requests. I'm testing with a role that has `read` access to `.lists-*` index and also has `endpoint_exceptions_all` kibana privilege, such as
```json5
{
    "metadata": {},
    "elasticsearch": {
        "cluster": [],
        "indices": [
            {
                "names": [
                    ".items-*",
                    ".alerts-security.alerts-*"
                ],
                "privileges": [
                    "view_index_metadata",
                    "read",
                    "write",
                    "manage"
                ],
                "field_security": {
                    "grant": [
                        "*"
                    ],
                    "except": []
                },
                "allow_restricted_indices": false
            },
            {
                "names": [
                    ".lists-*"
                ],
                "privileges": [
                    "read",
                    "manage",
                    "view_index_metadata"
                ],
                "field_security": {
                    "grant": [
                        "*"
                    ],
                    "except": []
                },
                "allow_restricted_indices": false
            }
        ],
        "run_as": []
    },
    "kibana": [
        {
            "base": [],
            "feature": {
                "siem": [
                    "minimal_all",
                    "endpoint_list_all",
                    "endpoint_exceptions_all"                
                 ]
            },
            "spaces": [
                "*"
            ]
        }
    ]
}
```
which does restrict the role from doing any CUD operations on the UX as expected.

However, on the API side the role is able to do CUD operations, since endpoints exceptions are added to saved objects and now I'm wondering if we need index access for it at all? Or is there a way to restrict saved objects access using index privileges?

Reply:
@ashokaditya I think there's a lot of context here that I'm missing. Is there a ticket describing the behavior that you're trying to achieve here? Is this a bugfix that's triggered a permissions discussion?

Broadly, though: we typically are not proactive with ES permissions checking on the server, and instead attempt to perform the requested action as the user requesting it. If an error is raised within ES, we will surface that as part of the response.

Reply:
Thanks again for looking into this @rylnd . Apologies for not providing context. The idea is to restrict `endpoint exceptions` for serverless users. Sepecifically when a user doesn't have any `endpoint` PLI (essentials/complete) they should not be able to access `endpoint exceptions`. Here is the [first PR](https://github.com/elastic/kibana/pull/164107) that introduced this check.

After talking to my team, looks like we don't need the index privilege check at all, much like we don't for other endpoint artefacts (trusted apps, event filters etc.). We drive this purely via kibana privileges. So in the end, we don't need to use the `useListsPrivileges` for checking permissions to `.lists-*` indices.

Reply:
@paul-tavares @dasansol92 @semd ☝🏼

Reply:
@kqualters-elastic @yctercero looks like we might need to look into this a bit more. "Endpoint exceptions" is not similar to endpoint artefacts and so I'm not so sure if access to it should be similar to endpoint artefacts. I'm going to try and find a time so we can quickly discuss this offline.

Reply:
I spoke with @vitaliidm offline and looks like endpoint exceptions do not need `.lists-*` permissions.